File: src/Binary_Tree_Level_Order_Traversal.java
/**
 * Definition for binary tree
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ArrayList<ArrayList<Integer>> levelOrder(TreeNode root) {
        // Start typing your Java solution below
        // DO NOT write main() function
        ArrayList<ArrayList<Integer>> ret = new ArrayList<ArrayList<Integer>>();
        if (root==null) return ret;
        
        LinkedList<TreeNode> queue = new LinkedList<TreeNode>();
        LinkedList<TreeNode> temp = new LinkedList<TreeNode>();
        queue.offer(root);
        
        ArrayList<Integer> list = new ArrayList<Integer>();
        
        while(!queue.isEmpty()){
            TreeNode node = queue.poll();
            list.add(node.val);
            if (node.left!=null)    temp.offer(node.left);
            if (node.right!=null)   temp.offer(node.right);
            if (queue.isEmpty() && !temp.isEmpty()){
                queue = temp;
                temp = new LinkedList<TreeNode>();
                ret.add(list);
                list = new ArrayList<Integer>();
            }
        }
        
        ret.add(list);
        return ret;
    }
}

File: src/python/tree/binary-tree-level-order-traversal.py

# Definition for a  binary tree node
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    # @param root, a tree node
    # @return a list of lists of integers
    def levelOrder(self, root):
        ret = []
        cur_node_list= []
        
        if root is not None:
            cur_node_list.append(root)
        while cur_node_list:
            curRet = []
            child_node_list = []
            for node in cur_node_list:
                curRet.append(node.val)
                if node.left:
                    child_node_list.append(node.left)
                if node.right:
                    child_node_list.append(node.right)
            ret.append(curRet)
            cur_node_list = child_node_list
        return ret